content/kubermatic/master/tutorials/opa/_index.en.md

```diff
@@ -0,0 +1,149 @@
++++
+title = "Open Policy Agent (OPA) via UI"
+date = 2021-02-10T12:07:15+02:00
+weight = 10
+
++++
+
+## Open Policy Agent (OPA)
+
+[Open Policy Agent](https://www.openpolicyagent.org/) (OPA) is an open-source, general-purpose policy engine that unifies policy enforcement across the stack. We are integrating it with using [Gatekeeper](https://github.com/open-policy-agent/gatekeeper), which is an OPA's Kubernetes-native policy engine. More info about OPA and Gatekeeper can be read from their docs and tutorials.
+
+
+### Admin Panel for OPA Options
+
+As an admin you will find a few options in the `Admin Panel`. You can access this panel by clicking on the account icon on the top right and select `Admin Panel`.
+
+![Access Admin Panel](admin.png)
+
+In here you can see the `OPA Options` with two checkboxes attached.
+- `Enable by Default`: Set `OPA Integration` checkbox on cluster creation to enabled by default. 
+- `Enforce`: Enable to make users unable to edit the checkbox.
+
+![Admin OPA Options](admin-options.png)
+
+The Admin Panel also offers you the possibility to specify Constraint Templates.
+
+![Constraint Templates](admin-ct.png)
+
+
+### Constraint Templates
+
+Constraint Templates allow you to declare new Constraints. They are intended to work as a schema for constraint parameters and enforce their behavior.
+To add a new constraint template click on the `+` icon on the right. A new dialog will appear, where you can specify the spec of the template:
+
+![Add Constraint Template](admin-add-ct.png)
+
+Following example requires all labels that are described by the constraint to be present:
+```
+crd:
+  spec:
+    names:
+      kind: K8sRequiredLabels
+    validation:
+      # Schema for the `parameters` field
+      openAPIV3Schema:
+        properties:
+          labels:
+            type: array
+            items: string
+targets:
+  - target: admission.k8s.gatekeeper.sh
+    rego: |
+      package k8srequiredlabels
+
+      violation[{"msg": msg, "details": {"missing_labels": missing}}] {
+        provided := {label | input.review.object.metadata.labels[label]}
+        required := {label | label := input.parameters.labels[_]}
+        missing := required - provided
+        count(missing) > 0
+        msg := sprintf("you must provide labels: %v", [missing])
+      }
+```
+
+Just click on `Add` to create the constraint template. In this table you can also edit or delete it again if needed.
+
+![Constraint Templates](admin-ct-overview.png)
+
+
+### Activating OPA during Cluster Creation
```

Comment:
i would put the OPA activating options together, and OPA operating stuff together. Now there is not a clear connection between the Constraint Templates and the Constraints.